dependencies.yaml

```
@@ -51,6 +57,14 @@ dependencies:
               cuda: "11.8"
             packages:
               - cudatoolkit=11.8
+  docs_build:
+    common:
+      - output_types: [requirements]
+        packages:
+          - sphinx
+          - sphinx_rtd_theme
+          - sphinx_copybutton
+          - sphinxcontrib-moderncmakedomain
```

```suggestion
      - output_types: [conda]
        packages:
          - pip
          - pip:
            - sphinxcontrib-moderncmakedomain
          - sphinx
          - sphinx-copybutton
          - sphinx_rtd_theme
```

As shown on the conda doc page below, we can actually add `pip` packages to a conda environment file like this.

This syntax is supported by  rapids-dependency-file-generator.

- https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually

Let's try to favor conda dependencies whenever possible for consistency with our other docs build jobs.